src/main/java/com/develhope/spring/order/service/OrderService.java

```diff
@@ -12,9 +16,22 @@
 public class OrderService {
     @Autowired
     private OrderRepository orderRepository;
+    @Autowired
+    private VehicleRepository vehicleRepository;
 
-    public Order createOrder(Order order) {
-        return orderRepository.save(order);
+    public Order shopVehicle(Vehicle vehicle, VehicleState isPurchasable, User userId) {
```

Comment:
oltre a ciò che ha fatto notare Mattia; perchè passi il vehicle state come parametro in ingresso quando gli passi già vehicle da cui te lo puoi ricavare?